archive/overlap.go

```diff
@@ -32,9 +36,11 @@ func alignChunksToSpans(chunks []Chunk, order zbuf.Order, filter nano.Span) []Sp
 				// last timestamp was just before ts.
 				siSpan := firstLastToSpan(siFirst, prevTs(ts, order))
 				if filter.Overlaps(siSpan) {
+					chunks := copyChunks(siChunks, nil)
+					chunksSort(order, chunks)
```

Comment:
Did you add the sort here for correctness? i.e. was the code doing the right thing prior to the sort being added?

Reply:
Before this PR, and specifically the 214c3712b7cc06ad46c204218d80f13db1270909 commit, the `boundaries` function returned chunks in a predictable order, so this change ensures that is maintained. I don't think this is needed for correctness, since the code that actually uses the chunks for data operations (like newSpanScanner) works on them as a group, and performs ordering based on the records that come back. Having a predictable order on the chunks is useful for things likes tests, where we can rely on the ordering for comparisons when we're listing the chunks in the archive, though.